internal/server/server.go

```diff
@@ -0,0 +1,287 @@
+package server
+
+import (
+	"bytes"
+	"errors"
+	"net/http"
+
+	dbstorage "github.com/developerc/project_gophermart_2/internal/db_storage"
+	"github.com/developerc/project_gophermart_2/internal/general"
+	"github.com/developerc/project_gophermart_2/internal/middleware"
+	"github.com/go-chi/chi/v5"
+	"github.com/jackc/pgerrcode"
+	"github.com/jackc/pgx/v5/pgconn"
+)
+
+type svc interface {
+	Register(buf bytes.Buffer) (*http.Cookie, error)
+	UserLogin(buf bytes.Buffer) (*http.Cookie, error)
+	GetUserFromCookie(cookieValue string) (string, error)
+	PostUserOrders(usr string, buf bytes.Buffer) error
+	GetUserOrders(usr string) ([]byte, error)
+	GetUserBalance(usr string) ([]byte, error)
+	PostBalanceWithdraw(usr string, buf bytes.Buffer) error
+	GetUserWithdrawals(usr string) ([]byte, error)
+}
+
+type Server struct {
+	service svc
+}
+
+func (s *Server) Register(w http.ResponseWriter, r *http.Request) {
+	var buf bytes.Buffer
+	var err error
+	_, err = buf.ReadFrom(r.Body)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	cookie, err := s.service.Register(buf)
+	if err != nil {
+		var pgErr *pgconn.PgError
+		switch {
+		case errors.As(err, &pgErr) && pgerrcode.IsIntegrityConstraintViolation(pgErr.Code) && pgErr.ConstraintName == "must_be_different_usr":
+			http.Error(w, err.Error(), http.StatusConflict)
+			return
+		default:
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+	}
+	http.SetCookie(w, cookie)
+
+	w.WriteHeader(http.StatusOK)
+}
+
+func (s *Server) UserLogin(w http.ResponseWriter, r *http.Request) {
+	var buf bytes.Buffer
+	var err error
+	_, err = buf.ReadFrom(r.Body)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	cookie, err := s.service.UserLogin(buf)
+	if err != nil {
+		if _, ok := err.(*dbstorage.ErrorLgnPsw); ok {
+			http.Error(w, err.Error(), http.StatusUnauthorized)
+			return
+		}
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	http.SetCookie(w, cookie)
+	w.WriteHeader(http.StatusOK)
+}
+
+func (s *Server) PostUserOrders(w http.ResponseWriter, r *http.Request) {
+	var usr string
+	var buf bytes.Buffer
+	cookie, err := r.Cookie("user")
+	if err != nil {
+		switch {
+		case errors.Is(err, http.ErrNoCookie):
+			http.Error(w, err.Error(), http.StatusUnauthorized)
+			return
+		default:
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+	}
+	usr, err = s.service.GetUserFromCookie(cookie.Value)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	_, err = buf.ReadFrom(r.Body)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	err = s.service.PostUserOrders(usr, buf)
```

Comment:
```suggestion
	err = s.service.PostUserOrders(r.Context(), usr, buf)
```
И так для всех функций, которым нужен в той или иной степени контекст.

Reply:
Исправил